### PR TITLE
dragmap: init at 1.3.0

### PR DESCRIPTION
**Pull request by apraga**
This is a resubmission of #307720 due to a massive ping.

Genomic mapper. Open-source version of DRAGEN.
Tests are run during execution with google-tests.

## Description of changes

Bioinformatics tools for genomics : align data according to a reference genome. Open-source version of proprietary software (DRAGEN).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) -
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

~The package builds fine on its own but fails with nix-pkgs review~ -> **This has been fixed**.

Thanks,
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/dr/dragmap/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  # Add missing include cstdint
+  # Upstream does not accept PR. Issue opened at
+  # https://github.com/Illumina/DRAGMAP/issues/63
+  patches = [ ./cstdint.patch ];
+
+  GTEST_INCLUDEDIR = "${gtest.dev}/include";
+  CPPFLAGS = "-I ${boost.dev}/include";
+  LDFLAGS = "-L ${boost.out}/lib";
+
+  installPhase = ''
```
Comment (drupol) [inline pkgs/by-name/dr/dragmap/package.nix:38]: Shouldn't this be the `postInstall` ?

File: pkgs/by-name/dr/dragmap/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    mkdir -p $out/bin
+    cp build/release/dragen-os $out/bin/
+
+    runHook postInstall
+  '';
+
+  # Tests are launched by default from makefile
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/dr/dragmap/package.nix:50]: no `meta.mainProgram` ?

**Comment by apraga** [comment]
Note: the new version pass nix-pkgs review.

**Comment by apraga** [comment]
@drupol Could you check out my answers to your questions and see if it answers them ? Thanks

**Comment by apraga** [comment]
@drupol If you have some time to look at this PR, it would be great, thanks !

File: pkgs/by-name/dr/dragmap/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  stdenv,
+  fetchFromGitHub,
+  boost,
+  cmake,
+  gtest,
+  zlib,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "DRAGMAP";
```
Comment (drupol) [inline pkgs/by-name/dr/dragmap/package.nix:12]: Why uppercase ?

**Review by drupol — CHANGES_REQUESTED** [review]
The commit log message **and** the PR title must be formatted according to the contributing guidelines. See the CONTRIBUTING guide.

File: pkgs/by-name/dr/dragmap/package.nix (hunk 1 above)
Comment (drupol) [reply pkgs/by-name/dr/dragmap/package.nix:43]: I got no feedback on this so far.

**Comment by drupol** [comment]
Moving this to draft until everything is fixed.

File: pkgs/by-name/dr/dragmap/package.nix (hunk 1 above)
Comment (apraga) [reply pkgs/by-name/dr/dragmap/package.nix:43]: Hi, thanks for the follow-up. All this time, my answers were marked as pending so it looks like you never saw it !
Updating the previous answer: if you are asking about the FLAGS, they are needed for compilation. The part in installPhase cannot be done in postInstall (as it tries to install things in /usr/bin).

**Comment by apraga** [comment]
Okay, I hope you are seeing the previous comments now ! PR title, commit have been reformatted and the additional commit have been removed.

**Review by apraga — COMMENTED** [review]
@drupol Hi, are you able to see my comments ? I struggled with github interface (comments were "stuck" into reviewing).

File: pkgs/by-name/dr/dragmap/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    # pclose is called on a NULL value. This is no longer allowed since
+    #  https://github.com/bminor/glibc/commit/64b1a44183a3094672ed304532bedb9acc707554
+    ./stdio-pclose.patch
+
+    # Add missing include cstdint.  Upstream does not accept PR. Issue opened at
+    # https://github.com/Illumina/DRAGMAP/issues/63
+    ./cstdint.patch
+  ];
+
+  GTEST_INCLUDEDIR = "${gtest.dev}/include";
```
Comment (drupol) [inline pkgs/by-name/dr/dragmap/package.nix:39]: These env. vars should be under the `env` attribute.

**Review by drupol — APPROVED** [review]

**Pull request merged**